Clean up the managed directories to avoid external side effect influence

exp-hc opened the pull request:
To avoid issue #14, I'm proposing using rsync to copy the new collection content and then deleting the original temp directory instead of using mv. I also found the title of this block of code did not represent the action.

webknjaz commented on action.yml:
```diff
@@ -156,16 +156,19 @@ runs:
       collection to ${{ steps.collection-metadata.outputs.checkout-path }}...
     shell: bash
-  - name: Install ansible-core (${{ inputs.ansible-core-version }})
+  - name: Move collection to test directory
```
I'd accept this in a separate PR.

webknjaz commented on action.yml:
```diff
@@ -156,16 +156,19 @@ runs:
       "${{ steps.collection-metadata.outputs.collection-namespace-path }}"
       ;
-      mv -v
+      rsync -a
```
We shouldn't support working in a dirty working dir. I was about to suggest doing the cleanup in the `post` stage (https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#post) but it looks like only JS actions support that: https://github.com/actions/runner/issues/1478.

So for now, I'd go for `rm -rf .tmp-ansible-collection-checkout/` as the first step.

exp-hc replied:
We would need to `rm -rf "${{ steps.collection-metadata.outputs.collection-namespace-path }}"` as the destination directory of the move is the issue. I can update this PR if that is the route you prefer.

exp-hc replied:
@webknjaz change pushed

webknjaz replied:
I'd prefer to clean up the whole directory we manage to prevent any similar problems with other dirs. Especially as we never expect external influence on that dir.

webknjaz replied:
As a side note: maybe I'll consider using `ln -sf` instead of `mv` in the future. But this is out of the scope here.

webknjaz commented on action.yml:
```diff
@@ -156,10 +156,13 @@ runs:
       set -x
       ;
+      rm -rf 
+      "${{ steps.collection-metadata.outputs.checkout-path }}"
```
This should:
1) Clean up the whole directory structure we manage to prevent any further side effects.
2) Do (1) as the very first step in this action, before the checkout.

webknjaz replied:
Moreover, I'd probably clean up both dirs, I think. Maybe the whole collections dir too.

exp-hc replied:
That makes sense, I've implemented the change per your suggestion.

exp-hc commented:
as another follow-up here, without removing the directories even a successful job will break subsequent jobs. 

### First job log output 
```
Run set -x ; mkdir -pv "ansible_collections/expedient" ; mv -v ".tmp-ansible-collection-checkout" "ansible_collections/expedient/smc" ; set +x
  set -x ; mkdir -pv "ansible_collections/expedient" ; mv -v ".tmp-ansible-collection-checkout" "ansible_collections/expedient/smc" ; set +x
  shell: /bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.9.9/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.9/x64/lib
+ mkdir -pv ansible_collections/expedient
mkdir: created directory 'ansible_collections'
mkdir: created directory 'ansible_collections/expedient'
+ mv -v .tmp-ansible-collection-checkout ansible_collections/expedient/smc
renamed '.tmp-ansible-collection-checkout' -> 'ansible_collections/expedient/smc'
+ set +x
```
### Second job log output
```
Run set -x ; mkdir -pv "ansible_collections/expedient" ; mv -v ".tmp-ansible-collection-checkout" "ansible_collections/expedient/smc" ; set +x
  set -x ; mkdir -pv "ansible_collections/expedient" ; mv -v ".tmp-ansible-collection-checkout" "ansible_collections/expedient/smc" ; set +x
  shell: /bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.9.9/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.9/x64/lib
+ mkdir -pv ansible_collections/expedient
+ mv -v .tmp-ansible-collection-checkout ansible_collections/expedient/smc
renamed '.tmp-ansible-collection-checkout' -> 'ansible_collections/expedient/smc/.tmp-ansible-collection-checkout'
+ set +x
```
